src/docs/csharp/best-practices/code-style.md

```diff
@@ -0,0 +1,82 @@
+---
+tags:
+  - melhores práticas
+  - testes
+--- 
+# Testes
+<authors>
+    <author user="andresantarosa"/> 
+</authors>  
+
+---
+Seguir um padrão de código é essencial para facilitar a legibilidade do código e a manutentabilidade da aplicação como um todo. Em geral linguagens possuem alguns padrões definidos de como codificar certos comandos afim de padronizar a escrita entre os desenvolvedores.
+
+## var
+O `var` é uma keywork utilizada na tipagem implícita, utilizando o `var` torna-se desncessário a explitação do tipo da variável e é uma grande aliada para deixar o código mais limpo e fluído. Entretanto é necessário ter cuidado na sua utilização, visto que se utilizado fora de lugar pode piorar consideravelmente a inteligibilidade.
+
+### Não utilizar `var` quando o tipo de retorno não é explícito.
+
```

Comment:
I disagree with that, it's possible use var in all scenarios.

Using var in this cases should the code keeps a pattern, but it's my opinion.

Reply:
I agree with @isaacnborges-ac ... only situation I don't use `var` is in cases like:

```csharp
string oi;
if (true)
   oi = "oi";
else
   oi = "ola";
```

Reply:
I prefer use var only when the return is explicit, makes de code a little bit easier to read, but it's also my opinion.

Reply:
I also think like that

// Bad
`var customer = new List<Customer>();`

// Good
`var customers = new List<Customer>()`;

Reply:
I agree with @leticiasassaki , `var` should be used sparingly and alwaysalongside with other best practices like name things properly and be as explicit as it could get.

@isaacnborges-ac  using `var` with a non explicit definition ends up turning the code a lot harder to read and lead to missinterpretations, I got a case like this some days ago in a declaration statement, the property name was Product but it was receiving an Order, as it was using `var` I relied on the property name since the right side of the expression was a little fuzzy. Nothing was making sense, then I hovered the mouse and saw the correct variable type (this after some brain damage trying to understand the code). I agree that using `var` the code ends up gaining some beautiness but also can lead to a lack os expressivity when things are not clearly as it should or it could.

Reply:
Exactly @andresantarosajsm. but you agree if the variable's name make sense, using `var` shouldn't be a problem. 
The point for us I guess is the code should be easy to understand.

Reply:
@isaacnborges-ac actually I dont like to relly on the variable name, I believe that the right side of the expression should explicit which type the variable will be.

Reply:
@isaacnborges-ac Thanks for pointing this repo, I'll add as reference for the next collaborations :)

Reply:
Well, good discussion, but it's time to determine if `var` is welcome or not haha. I vote in massive `var` use hahahaha

Reply:
🔝 Well, I'm in with @gabrielfaradaycadmus - faragod hahaha 😁

Reply:
So all I've seen of problems with `var` are actually variable name problems.
If it's to vote, I vote to use `var`.
Many languages don't even use var, and no one has a problem with that, so this shouldn't be an argument for the "Don't use var" defense

Reply:
I agree with @ricardochaves what we need to do is choose better variable names, and better method names, because even if you put the type to the left of variable declaration, what about other lines of code where that variable will be used? you don't have the type there.

Reply:
@ricardochaves we could have for and example a method that we call GetNamesAndAge(), this method could return either an List<Tuple>, Tuple[] or even a Dictionary<>, naming convention wouldn't allow us to know the return type unless we use GetNamesAndAgeAsArray or GetNamesAndAgeAsList

Reply:
but man, you really need to knows its type? For me it does not matter if it's an array or a list or an IEnumerable ... it's a collection, besides that if you have to return 2 different things, you may be wrong in your design, but ok, I see situation where it's necessary. And in that case you can return a tuple like `(names, ages)`